src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1387,6 +1387,16 @@ because xbuild doesn't support framework reference assemblies.
       ResourceName="MonoRuntimeProvider.Bundled.java"
       OutputPath="$(_AndroidIntermediateJavaSourceDirectory)mono\MonoRuntimeProvider.java"
   />
+  <CopyResource
+      Condition=" '$(_AndroidRuntime)' == 'NativeAOT' "
+      ResourceName="NativeAotRuntimeProvider.java"
+      OutputPath="$(_AndroidIntermediateJavaSourceDirectory)net\dot\jni\nativeaot\NativeAotRuntimeProvider.java"
+  />
+  <CopyResource
+      Condition=" '$(_AndroidRuntime)' == 'NativeAOT' "
+      ResourceName="JavaInteropRuntime.java"
+      OutputPath="$(_AndroidIntermediateJavaSourceDirectory)net\dot\jni\nativeaot\JavaInteropRuntime.java"
+  />
```

Comment:
TODO replace this line with main assembly name:

https://github.com/dotnet/android/blob/b7e21ad40bed68db17065734fcf31ee6d6e0c1ae/samples/NativeAOT/JavaInteropRuntime.java#L7-L8